.pre-commit-config.yaml

```diff
@@ -0,0 +1,23 @@
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.6.0
+  hooks:
+  - id: check-json
+  - id: check-toml
+  - id: check-yaml
+  - id: end-of-file-fixer
+  - id: trailing-whitespace
+
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  rev: v0.6.7
+  hooks:
+  - id: ruff
+    args: [--fix, --exit-non-zero-on-fix]
+  - id: ruff-format
+
+- repo: https://github.com/pre-commit/mirrors-mypy
+  rev: v1.11.2
+  hooks:
+  - id: mypy
+    additional_dependencies:
+    - types-requests
```

Comment:
Do we need this? This package doesn't use `requests` directly AFAICT.

Reply:
probably not, I just copied this over from target-snowflake's pre commit config file. will remove.

Reply:
Oh, sorry about that 😅